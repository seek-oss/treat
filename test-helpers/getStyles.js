export default async (page, selector) => {
  await page._client.send('DOM.enable');
  await page._client.send('CSS.enable');
  const doc = await page._client.send('DOM.getDocument');
  const { nodeId } = await page._client.send('DOM.querySelector', {
    nodeId: doc.root.nodeId,
    selector,
  });

  const styleForSingleNode = await page._client.send(
    'CSS.getMatchedStylesForNode',
    { nodeId },
  );

  return styleForSingleNode.matchedCSSRules.reduce((prev, curr) => {
    const styles = Object.assign(
      ...curr.rule.style.cssProperties.map((rule) => ({
        [rule.name]: rule.value,
      })),
    );

    return { ...prev, ...styles };
  }, {});
};
