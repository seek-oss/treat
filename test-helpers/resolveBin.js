import * as path from 'path';

export default (packageName, binName) => {
  const packageJson = require(`${packageName}/package.json`);
  const binPath =
    typeof packageJson.bin === 'string'
      ? packageJson.bin
      : packageJson.bin[binName || packageName];

  return require.resolve(path.join(packageName, binPath));
};
