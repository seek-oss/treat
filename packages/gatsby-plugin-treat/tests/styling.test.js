import execa from 'execa';
import * as path from 'path';
import waitForLocalhost from 'wait-for-localhost';
import getStyles from '../../../test-helpers/getStyles';
import resolveBin from '../../../test-helpers/resolveBin';

const gatsbyServerPort = 9999;
const gatsbyBinaryPath = resolveBin('gatsby', 'gatsby');
const gatsbyFixturePath = path.resolve(__dirname, '../../gatsby-plugin-treat-example');
const gatsbyExecArgs = {
  shell: '/bin/bash', 
  cwd: gatsbyFixturePath,
  env: {
    NODE_ENV: 'development'
  },
  extendEnv: true
};
let gatsbyProcess;

async function navigateToServerWhenReady() {
  await waitForLocalhost({port: gatsbyServerPort});
  await page.goto(`http://localhost:${gatsbyServerPort}/`);
}

async function startDevServer() {
  gatsbyProcess = execa(`${gatsbyBinaryPath} develop -p ${gatsbyServerPort}`, gatsbyExecArgs);
  gatsbyProcess.stdout.pipe(process.stdout);
  gatsbyProcess.stderr.pipe(process.stdout);
  await navigateToServerWhenReady();
}

async function startProdServer() {
  await execa(`${gatsbyBinaryPath} build`, gatsbyExecArgs);
  gatsbyProcess = execa(`${gatsbyBinaryPath} serve -p ${gatsbyServerPort}`, gatsbyExecArgs);
  gatsbyProcess.stdout.pipe(process.stdout);
  gatsbyProcess.stderr.pipe(process.stdout);
  await navigateToServerWhenReady();
}

async function stopServer() {
  gatsbyProcess.kill();
  await gatsbyProcess;
}

describe('gatsby', () => {

  describe('develop', () => {

    beforeAll(() => startDevServer());
    afterAll(() => stopServer());

    test('it renders some global styles', async () => {
      const styles = await getStyles(page, 'body');
      expect(styles).toMatchSnapshot();
    });
    
    test('it renders some button styles', async () => {
      await page.waitForSelector('button');
      const styles = await getStyles(page, 'button');
      expect(styles).toMatchSnapshot();
    });
    
    test('it loads and renders some span styles', async () => {
      await page.waitForSelector('button');
      await page.click('button');
      await page.waitForSelector('span');
      const styles = await getStyles(page, 'span');
      expect(styles).toMatchSnapshot();
    });

  });
  
  describe('build', () => {
    
    beforeAll(() => startProdServer());
    afterAll(() => stopServer());

    test('it renders some global styles', async () => {
      const styles = await getStyles(page, 'body');
      expect(styles).toMatchSnapshot();
    });
    
    test('it renders some button styles', async () => {
      await page.waitForSelector('button');
      const styles = await getStyles(page, 'button');
      expect(styles).toMatchSnapshot();
    });
    
    test('it loads and renders some span styles', async () => {
      await page.waitForSelector('button');
      await page.click('button');
      await page.waitForSelector('span');
      const styles = await getStyles(page, 'span');
      expect(styles).toMatchSnapshot();
    });

  });

});
