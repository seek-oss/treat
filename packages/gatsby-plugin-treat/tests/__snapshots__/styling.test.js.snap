// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby build it loads and renders some span styles 1`] = `Object {}`;

exports[`gatsby build it renders some button styles 1`] = `
Object {
  "-webkit-appearance": "button",
  "-webkit-writing-mode": "horizontal-tb !important",
  "align-items": "flex-start",
  "background-color": "#fff",
  "border": "2px solid #ff1493",
  "border-bottom-color": "rgb(255, 20, 147)",
  "border-bottom-left-radius": "6px",
  "border-bottom-right-radius": "6px",
  "border-bottom-style": "solid",
  "border-bottom-width": "2px",
  "border-image-outset": "initial",
  "border-image-repeat": "initial",
  "border-image-slice": "initial",
  "border-image-source": "initial",
  "border-image-width": "initial",
  "border-left-color": "rgb(255, 20, 147)",
  "border-left-style": "solid",
  "border-left-width": "2px",
  "border-radius": "6px",
  "border-right-color": "rgb(255, 20, 147)",
  "border-right-style": "solid",
  "border-right-width": "2px",
  "border-top-color": "rgb(255, 20, 147)",
  "border-top-left-radius": "6px",
  "border-top-right-radius": "6px",
  "border-top-style": "solid",
  "border-top-width": "2px",
  "box-sizing": "border-box",
  "color": "#ff1493",
  "cursor": "pointer",
  "display": "inline-block",
  "font-family": "-apple-system,BlinkMacSystemFont,Roboto,Oxygen,Ubuntu,Cantarell,“Fira Sans”,“Droid Sans”,“Helvetica Neue”,Arial,sans-serif",
  "font-size": "1em",
  "font-stretch": "normal",
  "font-style": "normal",
  "font-variant-caps": "normal",
  "font-variant-east-asian": "normal",
  "font-variant-ligatures": "normal",
  "font-variant-numeric": "normal",
  "font-weight": "400",
  "letter-spacing": "normal",
  "line-height": "normal",
  "margin-bottom": "0em",
  "margin-left": "0em",
  "margin-right": "0em",
  "margin-top": "0em",
  "padding": ".5em 1em",
  "padding-bottom": "0.5em",
  "padding-left": "1em",
  "padding-right": "1em",
  "padding-top": "0.5em",
  "text-align": "center",
  "text-indent": "0px",
  "text-rendering": "auto",
  "text-shadow": "none",
  "text-transform": "none",
  "word-spacing": "normal",
}
`;

exports[`gatsby build it renders some global styles 1`] = `
Object {
  "display": "block",
  "margin": "1em",
  "margin-bottom": "1em",
  "margin-left": "1em",
  "margin-right": "1em",
  "margin-top": "1em",
  "padding": "0",
  "padding-bottom": "0px",
  "padding-left": "0px",
  "padding-right": "0px",
  "padding-top": "0px",
}
`;

exports[`gatsby develop it loads and renders some span styles 1`] = `Object {}`;

exports[`gatsby develop it renders some button styles 1`] = `
Object {
  "-webkit-appearance": "button",
  "-webkit-writing-mode": "horizontal-tb !important",
  "align-items": "flex-start",
  "background-color": "white",
  "border": "2px solid deeppink",
  "border-bottom-color": "deeppink",
  "border-bottom-left-radius": "6px",
  "border-bottom-right-radius": "6px",
  "border-bottom-style": "solid",
  "border-bottom-width": "2px",
  "border-image-outset": "initial",
  "border-image-repeat": "initial",
  "border-image-slice": "initial",
  "border-image-source": "initial",
  "border-image-width": "initial",
  "border-left-color": "deeppink",
  "border-left-style": "solid",
  "border-left-width": "2px",
  "border-radius": "6px",
  "border-right-color": "deeppink",
  "border-right-style": "solid",
  "border-right-width": "2px",
  "border-top-color": "deeppink",
  "border-top-left-radius": "6px",
  "border-top-right-radius": "6px",
  "border-top-style": "solid",
  "border-top-width": "2px",
  "box-sizing": "border-box",
  "color": "deeppink",
  "cursor": "pointer",
  "display": "inline-block",
  "font-family": "-apple-system, BlinkMacSystemFont, Roboto, Oxygen, Ubuntu, Cantarell, “Fira Sans”, “Droid Sans”, “Helvetica Neue”, Arial, sans-serif",
  "font-size": "1em",
  "font-stretch": "normal",
  "font-style": "normal",
  "font-variant-caps": "normal",
  "font-variant-east-asian": "normal",
  "font-variant-ligatures": "normal",
  "font-variant-numeric": "normal",
  "font-weight": "400",
  "letter-spacing": "normal",
  "line-height": "normal",
  "margin-bottom": "0em",
  "margin-left": "0em",
  "margin-right": "0em",
  "margin-top": "0em",
  "padding": "0.5em 1em",
  "padding-bottom": "0.5em",
  "padding-left": "1em",
  "padding-right": "1em",
  "padding-top": "0.5em",
  "text-align": "center",
  "text-indent": "0px",
  "text-rendering": "auto",
  "text-shadow": "none",
  "text-transform": "none",
  "word-spacing": "normal",
}
`;

exports[`gatsby develop it renders some global styles 1`] = `
Object {
  "display": "block",
  "margin": "1em",
  "margin-bottom": "1em",
  "margin-left": "1em",
  "margin-right": "1em",
  "margin-top": "1em",
  "padding": "0",
  "padding-bottom": "0px",
  "padding-left": "0px",
  "padding-right": "0px",
  "padding-top": "0px",
}
`;
